teleport-server/instance.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_instance" "teleport_instance" {
```

Comment:
Why don't you use the instance module here?

Reply:
Because I wanted to have more control on the resources, and setting up an instance is simple enough to just do it without the module.
But more importantly, the instance module ignores changes in the user data, and this is a problem here, as we want to recycle the instance in case the user data changes.

Reply:
The reason why I am asking this is because I think we should always try to use our modules as much as possible and I don't see anything special here. We can make the `user-data` ignore as optional I think and keeping up-to-date 2 ebs_optimized list is a bit more complicated.

Reply:
ok, going to try if I can make that optional

Reply:
was too good to be true 😛 
```
Error: aws_instance.instance: lifecycle ignore_changes cannot contain interpolations
```

Reply:
So, apparently neither `lifecycle` nor `ignore_changes` blocks can contain interpolations, so we can't make that optional in the instance module. See this Terraform issue for more info: https://github.com/hashicorp/terraform/issues/3116

Reply:
👍